core/include/vecmem/containers/array.hpp

```diff
@@ -37,7 +36,8 @@ namespace vecmem {
    /// it does provide an optional second template argument, which can be used
    /// to set the fixed size of the array at compile time.
    ///
-   template< typename T, std::size_t N = details::array_invalid_size >
+   template< typename T, std::size_t N = details::array_invalid_size,
+             typename Allocator = vecmem::polymorphic_allocator< T > >
```

Comment:
`vecmem::array` is meant to be able to work on top of memory allocated through `vecmem::memory_resource`. That's why I chose to use `vecmem::polymorphic_allocator`. As I hoped that it would take care of constructing/destructing the array elements properly, even for complex types.

Using the full-blown `vecmem::polymorphic_allocator` is indeed a bit of an overkill. Since a `vecmem::array` object will issue exactly one allocation and one de-allocation call to the allocator. I just wanted to inherit the logic for doing this properly for complex types from `vecmem::polymorphic_allocator`.

Reply:
I'm realize now that the answer is (probably) arrays with dynamic size.